homeassistant/components/ssdp/__init__.py

```diff
@@ -146,6 +146,24 @@ def __getitem__(self, name: str) -> Any:
             return getattr(self, name)
         return self.upnp.get(name)
 
+    def get(self, name: str, default: Any = None) -> Any:
```

Comment:
@MartinHjelmare / @bdraco / @frenck
Should I add this also to the other `xxxServiceInfo`?
When I migrated the others using a `TypedDict` in the first pass, and then using a `dataclass` in the second pass it didn't seem required.
Now with hindsight I think it might cause issues with custom components (ie. breaking change before 2022.6)

Reply:
Where is the code that actually needs this?

Reply:
`grep "discovery_info.get(" homeassistant/components/**/**.py`

As an example:
https://github.com/home-assistant/core/blob/24687243785accb741af12d6e7a9bdd1041016a4/homeassistant/components/deconz/config_flow.py#L194-L197

Reply:
For zeroconf (and dhcp/usb/mqtt) they were fixed as I bumped into them.
I don't have example but I assume there will be some similar code in custom components.

Reply:
As much as I hate to have more temporary code, I think we should implement it in all places since someone will be doing it that way.

Reply:
I'll open a separate PR for the other discovery components.
This probably also applies for the `__iter__` implementation in #60339